File: 05.Data_structure/Sort/c/insertion_sort.c
#include <stdio.h>

void insertionSort(int arr[], int n) {
    for (int i = 1; i < n; i++) {
        int key = arr[i];
        int j = i - 1;
        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];  // 한 칸 밀기
            j--;
        }
        arr[j + 1] = key;         // 삽입
    }
}

int main() {
    int arr[] = {5, 1, 4, 2, 8};
    int n = sizeof(arr) / sizeof(arr[0]);
    insertionSort(arr, n);
    for (int i = 0; i < n; i++)
        printf("%d ", arr[i]);
    return 0;
}

File: 05.Data_structure/Sort/py/insertion_sort.py
def insertion_sort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]  # 한 칸 밀기
            j -= 1
        arr[j + 1] = key         # 삽입
    return arr

print(insertion_sort([5, 1, 4, 2, 8]))  # [1, 2, 4, 5, 8]
